From a5c6ee4b967f7a491b8fba46c7df5c9adc8e6db5 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 22 Feb 2024 10:16:44 +0100
Subject: [PATCH 0683/1695] plat-stm32mp1: conf: default enable PKA

Default enable PKA compilation.
Enable the STM32_CRYPTO_DRIVERS if PKA or CRYP is compiled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ff93271b666198147543bae2c62351a66d001f40 plat-stm32mp1: conf: default enable PKA]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: I650d940522c456ea734afceedb32741195cedceb
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362225
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 5d0fa4fb9..9dafaba6c 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -196,6 +196,7 @@ $(call force,CFG_HALT_CORES_ON_PANIC_SGI,15)
 $(call force,CFG_SCMI_MSG_PERF_DOMAIN,n)
 $(call force,CFG_SECONDARY_INIT_CNTFRQ,y)
 $(call force,CFG_STM32MP1_SHARED_RESOURCES,y)
+$(call force,CFG_STM32_PKA,n)
 $(call force,CFG_STM32_SAES,n)
 $(call force,CFG_STM32MP1_RSTCTRL,y)
 $(call force,CFG_STM32MP15_CLK,y)
@@ -280,6 +281,7 @@ CFG_STM32_HASH ?= y
 CFG_STM32_I2C ?= y
 CFG_STM32_IWDG ?= y
 CFG_STM32_LPTIMER ?= y
+CFG_STM32_PKA ?= y
 CFG_STM32_RNG ?= y
 CFG_STM32_RSTCTRL ?= y
 CFG_STM32_RTC ?= y
@@ -317,7 +319,7 @@ endif # cfg-all-enabled, CFG_STM32_CRYP CFG_STM32_SAES
 endif # CFG_STM32MP13
 
 # If any crypto driver is enabled, enable the crypto-framework layer
-ifeq ($(call cfg-one-enabled, CFG_STM32_CRYP CFG_STM32_SAES CFG_STM32_HASH),y)
+ifeq ($(call cfg-one-enabled, CFG_STM32_CRYP CFG_STM32_HASH CFG_STM32_PKA CFG_STM32_SAES ),y)
 $(call force,CFG_STM32_CRYPTO_DRIVER,y)
 endif
 
-- 
2.47.2

